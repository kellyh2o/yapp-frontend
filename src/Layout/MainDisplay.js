import React, { Component } from "react";
import { connect } from "react-redux";
import Dashboard from "../Dashboard/Components/Dashboard";
import UserProfile from "../Users/Components/UserProfile";
import Settings from "../Settings/Components/Settings";


class MainDisplay extends Component {

    render() {
        switch (this.props.selectedView) {
            case "Settings":
                return <Settings />;
            case "UserProfile":
                return <UserProfile />;
            case "Dashboard":
            default: 
                return <Dashboard />;
        }
    }
}

const mapStateToProps = (state) => {
    return {
        selectedView: state.selectedView,
    };
}

export default connect(mapStateToProps)(MainDisplay);
