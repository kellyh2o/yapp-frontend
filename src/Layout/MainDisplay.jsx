import React, { Component, FunctionComponent } from 'react';
import clsx from 'clsx';
import { makeStyles, useTheme } from '@material-ui/core/styles';
import Drawer from '@material-ui/core/Drawer';
import CssBaseline from '@material-ui/core/CssBaseline';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import List from '@material-ui/core/List';
import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import SelectedDisplay from './SelectedDisplay';
import { connect } from 'react-redux';
import LeftDrawerMenu from './LeftDrawerMenu';



const drawerWidth = 240;

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
  },
  appBar: {
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    background: '#039be5'
  },
  appBarShift: {
    width: `calc(100% - ${drawerWidth}px)`,
    marginLeft: drawerWidth,
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  hide: {
    display: 'none',
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
  },
  drawerPaper: {
    width: drawerWidth,
  },
  drawerHeader: {
    display: 'flex',
    alignItems: 'center',
    padding: theme.spacing(0, 1),
    // necessary for content to be below app bar
    ...theme.mixins.toolbar,
    justifyContent: 'center',
    background: '#039be5',
  },
  mainPanelHeader: {
    display: 'flex',
    alignItems: 'center',
    padding: theme.spacing(0, 1),
    // necessary for content to be below app bar
    ...theme.mixins.toolbar,
    justifyContent: 'center',
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing(3),
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    })
  },
  contentShift: {
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
    marginLeft: 0,
  },
}));

export default function MainDisplay() {
  const classes = useStyles();
  const theme = useTheme();
  const [open, setOpen] = React.useState(false);

  const handleDrawerOpen = () => {
    setOpen(true);
  };

  const handleDrawerClose = () => {
    setOpen(false);
  };

  return (
    <div className={classes.root}>
      <CssBaseline />
      <AppBar
        position="fixed"
        className={clsx(classes.appBar, {
          [classes.appBarShift]: open,
        })}
      >
        <Toolbar>
          <IconButton
            color="inherit"
            aria-label="open drawer"
            onClick={handleDrawerOpen}
            edge="start"
            className={clsx(classes.menuButton, open && classes.hide)}
          >
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" noWrap>
            Persistent drawer
          </Typography>
        </Toolbar>
      </AppBar>
      <Drawer
        className={classes.drawer}
        variant="persistent"
        anchor="left"
        open={true}
        classes={{
          paper: classes.drawerPaper,
        }}
      >
        <div className={classes.drawerHeader}>
          <Typography align='center' variant="h5">
            Y A P P
          </Typography>
        </div>
        <Divider />
            <LeftDrawerMenu />
      </Drawer>
      <main
        className={clsx(classes.content, {
          [classes.contentShift]: open,
        })}
      >
        <div className={classes.mainPanelHeader} />
        <SelectedDisplay />
      </main>
    </div>
  );
}
