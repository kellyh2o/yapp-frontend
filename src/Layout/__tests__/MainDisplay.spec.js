import { render } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import MainDisplay from "../MainDisplay";

describe('MainDisplay', () => {
    it("should render", () => {
        const mockStore = configureMockStore();

        const initialState = {
            demoLocation: {
                name: "Hogwarts",
                jumps: [ 
                    {
                        name: "Jump1",
                        events: []
                    },
                    {
                        name: "Jump2",
                        events: []
                    },
                    {
                        name: "Jump3",
                        events: [
                            {
                                riderMass: 22,
                                riderSpeed: 12,
                            }
                        ]
                    }
                ]
            }
        };
        const store = mockStore(initialState);

        const { asFragment } = render(
            <Provider store={store}>
                <MainDisplay />
            </Provider>
        );

        expect(asFragment()).toMatchSnapshot();
    })
});

