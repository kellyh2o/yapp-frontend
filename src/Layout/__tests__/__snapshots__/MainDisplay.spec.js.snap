// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainDisplay should render 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionAbsolute MuiAppBar-colorPrimary makeStyles-appBar-4 makeStyles-appBarShift-5 MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-2 MuiToolbar-gutters"
      >
        <button
          aria-label="open drawer"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-6 makeStyles-menuButtonHidden-7 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h1
          class="MuiTypography-root makeStyles-title-8 MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap"
        />
        <div
          class="makeStyles-root-110"
        >
          <div>
            <button
              aria-haspopup="true"
              class="MuiButtonBase-root MuiIconButton-root makeStyles-normalButton-112 makeStyles-normalButton-113 MuiIconButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </header>
    <div
      class="MuiDrawer-root MuiDrawer-docked"
    >
      <div
        class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-9 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
      >
        <div
          class="makeStyles-toolbarIcon-3"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <hr
          class="MuiDivider-root"
        />
        <ul
          class="MuiList-root MuiList-padding"
        >
          <div>
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Dashboard
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M5 9.2h3V19H5zM10.6 5h2.8v14h-2.8zm5.6 8H19v6h-2.8z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Reports
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </div>
        </ul>
      </div>
    </div>
    <main
      class="makeStyles-content-12"
    >
      <div
        class="makeStyles-appBarSpacer-11"
      />
      <div
        class="MuiContainer-root makeStyles-container-13 MuiContainer-maxWidthLg"
      >
        <div>
          <div
            class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
            style="margin-bottom: 20px; padding: 20px;"
          >
            <h2
              class="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
            >
              Location
            </h2>
            <p
              class="MuiTypography-root MuiTypography-h4"
              style="padding-top: 20px;"
            >
              Hogwarts
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 MuiGrid-grid-lg-9"
            >
              <div
                class="MuiPaper-root makeStyles-paper-164 makeStyles-fixedHeight-165 MuiPaper-elevation1 MuiPaper-rounded"
              >
                <h2
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
                >
                  Peak Times
                </h2>
                <div
                  class="recharts-responsive-container"
                  style="width: 100%; height: 100%;"
                >
                  <div
                    style="position: absolute; width: 0px; height: 0px; visibility: hidden; display: none;"
                  />
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
            >
              <div
                class="MuiPaper-root makeStyles-paper-164 makeStyles-fixedHeight-165 MuiPaper-elevation1 MuiPaper-rounded"
              >
                <h2
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
                >
                  Today's Most Popular Jump
                </h2>
                <p
                  class="MuiTypography-root MuiTypography-h4"
                  style="padding-top: 20px;"
                >
                  Jump3
                </p>
                <p
                  class="MuiTypography-root makeStyles-depositContext-269 MuiTypography-body1 MuiTypography-colorTextSecondary"
                  style="padding-top: 20px; padding-left: 5px;"
                >
                  1 Hits Today
                </p>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiPaper-root makeStyles-paper-164 MuiPaper-elevation1 MuiPaper-rounded"
              >
                <h2
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
                >
                  Recent Jump Events
                </h2>
                <table
                  class="MuiTable-root"
                >
                  <thead
                    class="MuiTableHead-root"
                  >
                    <tr
                      class="MuiTableRow-root MuiTableRow-head"
                    >
                      <th
                        class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                        scope="col"
                      >
                        Date
                      </th>
                      <th
                        class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                        scope="col"
                      >
                        Jump Name
                      </th>
                      <th
                        class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                        scope="col"
                      >
                        Rider Mass (lbs) 
                      </th>
                      <th
                        class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                        scope="col"
                      >
                        Rider Speed (mph) 
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="MuiTableBody-root"
                  >
                    <tr
                      class="MuiTableRow-root"
                    >
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                      >
                        Invalid date
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                      >
                        Jump3
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                      >
                        22
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                      >
                        12
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-292"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
          >
            Built by 
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
              href="https://github.com/kellyh2o"
            >
              @kellyh2o
            </a>
            .
          </p>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
